includes/razorpay-webhook.php

```diff
@@ -110,10 +144,162 @@ protected function paymentAuthorized($data)
             // If the merchant has enabled auto capture
             //
             $payment->capture(array('amount' => $amount));
+
+            $success = true;
+        }
+
+        $this->razorpay->updateOrder($order, $success, $errorMessage, $razorpayPaymentId, true);
+
+        exit;
+    }
+
+    /**
+     * Currently we handle only subscription failures using this webhook
+     *
+     * @param $data
+     */
+    protected function paymentFailed($data)
+    {
+        //
+        // Order entity should be sent as part of the webhook payload
+        //
+        $orderId = $data['payload']['payment']['entity']['notes']['woocommerce_order_id'];
```

Comment:
@captn3m0 I get that, but I mean, this code will be triggered only if webhook url is set to the woocommerce url. So I think we're fine here.